packages/pagination/src/Pagination.types.ts

```diff
@@ -0,0 +1,117 @@
+import React from 'react';
+
+export type CustomOnClick = (page: number) => void;
+
+export interface PaginationProps
+  extends Omit<React.HTMLAttributes<HTMLElement>, 'onClick'> {
+  /**
+   * `className` - additional styles, like color
+   */
+  className?: string;
+  /**
+   * `totalPages` - total number of pages
+   */
+  totalPages: number;
+  /**
+   * `currentPage` - current page
+   */
+  currentPage: number;
+  /**
+   * `renderItem` - template function to allow different components and props for PaginationItem
+   */
+  renderItem?: (item: PaginationItemProps) => JSX.Element;
+  /**
+   * `onClick` - template function for onClick event of items
+   */
+  onClick?: CustomOnClick;
+  /**
+   * `aria-label` - aria-label for root pagination component
+   */
+  'aria-label'?: string;
+  /**
+   * `getItemAriaLabel` - Function for generation of aria-label of clickable items
+   */
+  getItemAriaLabel?: getItemAriaLabelFunction;
+}
+
+export interface DefaultGetItemAriaLabelProps {
+  type: 'page' | 'next' | 'previous';
```

Comment:
In other cases, like in `PaginationItemProps`, it's pretty direct that the `type` is of the item itself. It is really generic, because each item can be a page, a button, or maybe even ellipsis.

For the page/next/previous, I think whomever use the pagination will understand it
<img width="170" alt="image" src="https://user-images.githubusercontent.com/12513049/218069949-a0125f1f-68c9-4875-92e0-9b6c7a99d934.png">